Homework7/FightLogic.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Homework7
+{
+    class FightLogic
+    {
+        public static Unit GenerateFighter()
+        {
+            int randomIndex = Randomizer.RandomInt(0, 3);
+            Unit unit;
+
+            switch (randomIndex)
+                {
+                case 0:
+                    unit = new Warrior();
+                        break;
+
+                case 1:
+                    unit = new Archer();
+                    break;
+
+                default:
+                    unit = new Wizard();
+                    break;
+            }
+
+            return unit;
+        }
+
+        public static Unit GenerateFighter(FighterType? fighterType)
+        {
+            Unit unit;
+
+            switch (fighterType)
+            {
+                case FighterType.Warrior:
+                    unit = new Warrior();
+                    break;
+
+                case FighterType.Archer:
+                    unit = new Archer();
+                    break;
+
+                case FighterType.Wizard:
+                    unit = new Wizard();
+                    break;
+
+                default:
+                    unit = GenerateFighter();
+                    break;
+            }
+
+            return unit;
+        }
+
+        public static Unit FightOneOnOne(FighterType? fighter1Type = null, FighterType? fighter2Type = null)
+        {
+            Unit unit1 = fighter1Type != null ? GenerateFighter(fighter1Type) : GenerateFighter();
+            Unit unit2 = fighter2Type != null ? GenerateFighter(fighter2Type) : GenerateFighter();
+
+            Unit winner;
+
+            while (true)
+            {
+                if (Randomizer.RandomBool())
+                {
+                    unit1.Defense(unit2);
+                }
+                else
+                {
+                    unit2.Defense(unit1);
+                }
+
+                if (!unit1.IsAlive)
+                {
+                    winner = unit2;
+                    break;
+                }
+
+
+                if (!unit2.IsAlive)
+                {
+                    winner = unit1;
+                    break;
+                }
+            }
+
+            return winner;
+        }
+
+        public static Group FightGroupByGroup(int countFightersPerGroup)
```

Comment:
Also bad method for FightLogic. Move to Main.
Why bad? - it:
1. generates units.
2. creating groups
3. attack group units with another groups with some logic.
4. detects winner.

Reply:
move some actions in separate methods